packages/cardano-services/src/NetworkInfo/DbSyncNetworkInfoProvider/DbSyncNetworkInfoProvider.ts

```diff
@@ -115,19 +113,13 @@ export class DbSyncNetworkInfoProvider extends DbSyncProvider implements Network
   }
 
   async start(): Promise<void> {
+    this.#epochPollService.start();
```

Comment:
If it's injected as per previous comment, then it should be started/shutdown where it's created instead of here

```suggestion
```

Reply:
I think we should keep the service's load within [serviceMapFactory](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/cardano-services/src/Program/loadHttpServer.ts#L56) pure, without sending DB operations (same reason that we removed `healthChecks` on service init), so probably keeping `this.#epochPollService.start()` invoked by [service startImpl()](https://github.com/input-output-hk/cardano-js-sdk/blob/cc82bc27539e3ff07f7c2d5816fa7e70c32d06ac/packages/cardano-services/src/Http/HttpService.ts#L41-L43) after initialization is fine?

Reply:
Agree. It's reasonable for the `NetworkInfoProvider` to maintain the lifecycle of this service.

Reply:
If NetworkInfoProvider controls the lifecycle, then I think it should be the one creating the service, which is not ideal as it can logically be a separate service.

How about **not** having an explicit start & shutdown, but instead polling if there is at least 1 callback registered? Would then also need a way to unsubscribe, like:
```ts
export type Unsubscribe = () => void;
export interface EpochMonitor {
  onEpochRollover(callback: (epoch: Cardano.Epoch) => void)): Unsubscribe;
}
```

Reply:
OK, yeah, that's a nice solution. Each consumer can then have control of the monitor's function within it's lifecycle without impacting the others. Could also be:

```ts
export type Disposer = () => void;
```
... but the pattern is the important aspect. Nice one @mkazlauskas

Reply:
If there's an intent of introducing rxjs then we could use a type that's compatible with [Unsubscribable](https://github.com/ReactiveX/rxjs/blob/5d0e41ee46337f56dba6d9a1ef938002b261ccac/src/internal/types.ts#L65)